dotcom-rendering/scripts/env/check-deps.js

```diff
@@ -1,9 +1,33 @@
+const fs = require('fs');
+const lockfile = require('@yarnpkg/lockfile');
 const pkg = require('../../package.json');
+const { warn, log } = require('./log');
 
 if (pkg.devDependencies) {
-	const { warn, log } = require('./log');
-
-	warn('Don’t use devDependencies');
+	warn('Don’t use dependencies');
 	log('See https://github.com/guardian/dotcom-rendering/pull/4001');
 	process.exit(1);
 }
+
+/** @type {{ object: import('@types/yarnpkg__lockfile').LockFileObject }}*/
+const { object: json } = lockfile.parse(
+	fs.readFileSync('../yarn.lock', 'utf8'),
+);
+
+const knownNonSemver = /** @type {const} */ ([
+	'https://github.com/guardian/babel-plugin-px-to-rem#v0.1.0',
+]);
+
+const mismatches = Object.entries(pkg.dependencies)
+	.filter(([name, version]) => {
+		const pinned = json[name + '@' + version]?.version;
+		return version !== pinned;
+	})
+	.filter(([, version]) => !knownNonSemver.includes(version));
+
+if (mismatches.length) warn('All dependencies should be pinned');
+for (const [name, version] of mismatches) {
+	warn(`You must fix: ${name}@${String(version)}`);
```

Comment:
Should probably be `console.error` rather than `warn` (it looks like this specific `warn` calls `console.log` under the hood) because the script will fail soon, and semantically errors should be printed to `stderr` instead of `stdout`